### PR TITLE
[1948] Update WebDAV Linux automount option documentation

### LINKED ISSUE #1948: Documentation/WebDAV mounting on Linux: Warn about usage of "auto" in /etc/fstab
In `docs/modules/user_manual/pages/files/access_webdav.adoc` there is a section "Creating WebDAV Mounts on the Linux Command Line", describing how to auto-mount owncloud to a Linux file system.

In (currently) line 168 it is recommended to use "auto" in `/etc/fstab`, so owncloud is mounted automatically.
At lines 182 to 184 it says:
"Now every time you login to your Linux system your ownCloud share should
automatically mount via WebDAV in your `~/owncloud` directory. If you
prefer to mount it manually, change `auto` to `noauto` in `/etc/fstab`."

This is not exactly correct, at least not for my freshly installed Debian Buster: Auto-mounting via fstab does not happen on **login**, but on **boot**, before login.

Why is this important? Because when using "auto", mounting owncloud seems to be part of the target "Local File System" in the boot process. And if mounting owncloud fails for some reason, the PC won't boot:

```
...
[FAILED] Failed to mount /home/[my username]/owncloud
[DEPEND] Dependency failed for Local File System
[  OK  ] Started Emergency Shell.
[  OK  ] Stopped Dispatch Password Request to Console Directory Watch
[  OK  ] Reached target Emergency Mode.
[FAILED] Failed to start Dispatch Password Request to Console Directory Watch
...

Cannot open access to console, the root account is locked.
Press Enter to continue.
```
On pressing Enter, the message just appears again, I had to poweroff the machine manually.

This is easily fixable (I just changed to "noauto" in `/etc/fstab`, using a live Ubuntu DVD), and it is quite straight-forward to understand why auto-mounting an online-resource on boot is not a good idea on a laptop without constant internet access, but I wish the documentation had warned me about this, or recommended "noauto" with a comment on the possible "auto" option.

At least the sentence "Now every time you login to your Linux system..." should be changed to "Now every time you boot your Linux system", if I understand this correctly.

I am no expert on these topics, sorry if I misunderstood something.

### PR DESCRIPTION
**Pull request by settermjd**
This PR, specifically 299e10b, fixes #1948, by adding an admonition that clearly lays out the
potential pitfalls of auto-mounting remote filesystems and the impact that they can have on Linux server boot-time. It also adds a compromise recommendation.

### CODE REVIEW

**Comment by settermjd** [comment]
@jnweiger, following your feedback, I've refined the text. Does it need further changes?

**Comment by settermjd** [comment]
ping @jnweiger

File: modules/user_manual/pages/files/access_webdav.adoc [hunk 1]
```diff
@@ -109,31 +109,30 @@ image:files/access_webdav/dolphin-network-folder-wizard.png[Dolphin Network Fold
...
-mount WebDAV shares just like any other remote filesystem. Use this
-command to install it on Debian/Ubuntu:
+. Install the `davfs2` WebDAV filesystem driver, which allows you to mount WebDAV shares just like any other remote filesystem. 
+  Use this command to install it on Debian/Ubuntu:
 +
 ----
 sudo apt-get install davfs2
 ----
-. Use this command to install it on CentOS, Fedora, and openSUSE:
+. Use this command to install it on _CentOS_, _Fedora_, and _openSUSE_:
```
Comment (jnweiger) [inline modules/user_manual/pages/files/access_webdav.adoc]: `yum install` is centos fedora and rhel.
opensuse has `zypper install`

Testing on 15.1:

```
zypper install davfs2
Building repository 'Non-OSS Repository' cache ...........................................................................................[done]
Building repository 'Main Repository' cache ..............................................................................................[done]
Building repository 'Main Update Repository' cache .......................................................................................[done]
Building repository 'Update Repository (Non-Oss)' cache ..................................................................................[done]
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following 4 NEW packages are going to be installed:
  davfs2 fuse libexpat1 libneon27

4 new packages to install.
```
yep!

**Comment by settermjd** [comment]
Thanks for the further feedback, @jnweiger.

File: modules/user_manual/pages/files/access_webdav.adoc [hunk 2]
```diff
@@ -173,9 +170,33 @@ mount ~/owncloud
...
+[  OK  ] Stopped Dispatch Password Request to Console Directory Watch
+[  OK  ] Reached target Emergency Mode.
+[FAILED] Failed to start Dispatch Password Request to Console Directory Watch
+...
+
+Cannot open access to console, the root account is locked.
+Press Enter to continue.
+----
+
+For that reason, it may be better to instead always set the partition to `noauto` and mount the partition toward the end of the boot process.
```
[the remark this answers is not in the document]
Comment (settermjd) [reply modules/user_manual/pages/files/access_webdav.adoc]: Not currently, no. I'll find a solution today.

File: modules/user_manual/pages/files/access_webdav.adoc (hunk 1 above)
Comment (settermjd) [reply modules/user_manual/pages/files/access_webdav.adoc]: Thanks for spotting that one. Have fixed in a local commit and will push shortly.

File: modules/user_manual/pages/files/access_webdav.adoc [hunk 3]
```diff
@@ -173,9 +170,33 @@ mount ~/owncloud
...
+To find out more information about mounting partitions toward the end of the boot process in either SysVInit or systemd, please refer to the links below:
+
+- https://fedoramagazine.org/systemd-unit-dependencies-and-order/[systemd: Unit dependencies and order]
+- https://en.wikipedia.org/wiki/Init#SYSV[SysVInit]
+====
```
[the remark this answers is not in the document]
Comment (settermjd) [reply modules/user_manual/pages/files/access_webdav.adoc]: I've been talking with people in #linux on irc and the most recommended option is systemd's automount. I can mount the filesystem manually and make any number of read and write changes without a problem. However, when attempting to automount the filesystem with systemd davfs seems to constantly core dump. I'm not sure why yet (all feedback welcome). But here's a snippet from /var/log/syslog:

```
Jan  8 15:10:42 settermjd-desktop systemd[1]: Mounting /home/settermjd/owncloud...
Jan  8 15:10:42 settermjd-desktop mount.davfs: davfs2 1.5.4
Jan  8 15:10:42 settermjd-desktop systemd-coredump[10191]: Process 10184 (mount.davfs) of user 0 dumped core.#012#012Stack trace of thread 10184:#012#0  0x00007fa2630c6e97 __GI_raise (libc.so.6)#012#1  0x00007fa2630c8801 __GI_abort (libc.so.6)#012#2  0x0000561fc968b040 n/a (mount.davfs)#012#3  0x0000561fc968025c n/a (mount.davfs)#012#4  0x00007fa2630a9b97 __libc_start_main (libc.so.6)#012#5  0x0000561fc9680309 n/a (mount.davfs)
Jan  8 15:10:42 settermjd-desktop systemd[1]: Started Process Core Dump (PID 10219/UID 0).
Jan  8 15:10:42 settermjd-desktop systemd[1]: home-settermjd-owncloud.mount: Mount process exited, code=exited status=255
Jan  8 15:10:42 settermjd-desktop systemd[1]: home-settermjd-owncloud.mount: Failed with result 'exit-code'.
Jan  8 15:10:42 settermjd-desktop systemd[1]: Failed to mount /home/settermjd/owncloud.
Jan  8 15:10:42 settermjd-desktop systemd[1]: home-settermjd-owncloud.automount: Got automount request for /home/settermjd/owncloud, triggered by 2517 (gvfs-udisks2-vo)
```

File: modules/user_manual/pages/files/access_webdav.adoc [hunk 4]
```diff
@@ -173,9 +170,33 @@ mount ~/owncloud
...
 ----
 
-Now every time you login to your Linux system your ownCloud share should
-automatically mount via WebDAV in your `~/owncloud` directory. If you
-prefer to mount it manually, change `auto` to `noauto` in `/etc/fstab`.
+Now, every time you boot your Linux system your ownCloud share will automatically mount, via WebDAV, in your `~/owncloud` directory. 
+
+[IMPORTANT]
+====
+If the share is set to mount automatically and (_for whatever reason_) your ownCloud server fails to start, your server will fail to properly boot.
```
[the remark this answers is not in the document]
Comment (settermjd) [reply modules/user_manual/pages/files/access_webdav.adoc]: I've updated this locally and will push the change later today.

File: modules/user_manual/pages/files/access_webdav.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (settermjd) [reply modules/user_manual/pages/files/access_webdav.adoc]: See the feedback below about davfs core dumping/segfaulting.

**Comment by michaelstingl** [comment]
@jnweiger the changes you requested were made. Ready to approve?

**Pull request closed without merging**